pages/docs/reason-compiler/latest/decorators.mdx

```diff
@@ -630,7 +639,16 @@ let actions = [
 
 ### `[@bs.deriving abstract]`
 
-Reason records are represented as arrays in JS:
+Reason records were represented as arrays in JS in BuckleScript versions < v7. `bs.deriving abstract` makes it more convenient to access records when you or your library consumers need to be on an older BuckleScript version.
```

Comment:
Ditto, we shouldn't focus on libraries etc, just explain the feature what it does, have the same short note that they might want to use records-as-objects instead.